lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -74,7 +73,11 @@ def pretty_timeout_type
 
     def deploy_timed_out?
       # Do not use the hard timeout if progress deadline is set
-      @progress_condition.present? ? deploy_failing_to_progress? : super
+      if @definition['spec']['progressDeadlineSeconds'].present? && @progress_condition.present?
```

Comment:
Looks like `k8s` merges in a default, this will check to see if you manually set one. This might be a behavior change

Reply:
Yes, it would be a behaviour change. Why do you think that would be better? IMO getting the smarter timeout logic shouldn't depend on whether you liked the default value the server gives you. (interestingly, we originally intended to look at `@definition` but made a mistake... and I ended up thinking it was a happy accident 😄 )

Reply:
I don't have strong feelings either way. Though the behavior being different on first vs subsequent deploys seems less than ideal and makes me wonder if there are cases the test aren't capturing. 

My bigger issue is without changing this behavior, I have no idea how to write a reliable test here.

Reply:
> the behavior being different on first vs subsequent deploys

What do you mean by this? Why would that be the case?

> My bigger issue is without changing this behavior, I have no idea how to write a reliable test here.

Maybe I'm missing the obvious, but why does changing this behaviour help? I agree this is tricky to test reliably... you'd almost want a way to prevent the whole RS from coming up... I'll think about it.